### PR TITLE
Remote Save Provider refactoring and enhancements 

### PR DESCRIPTION
**Pull request by vedg**
Added save entity detailed structure and enhancements

Requirements added here: https://github.com/microsoft/vscode-dev/issues/477

Testing:

npm run compile-web
tested with portal route - http://localhost:3000/powerplatform/portal/webpages/563a8e53-6cf7-ec11-82e4-000d3a8b4b81?orgUrl=org4e297bda.crm10.dynamics.com&source=dataverse&schema=portalDataverseoldschema&websiteId=24828659-6cf7-ec11-82e4-000d3a8b4b81

### CODE REVIEW

File: src/web/client/common/remoteFetchProvider.ts [hunk 1]
```diff
@@ -93,13 +94,13 @@ function createContentFiles(result: string, entity: string, queryParamsMap: any,
         vscode.window.showInformationMessage(patchRequestUrl)
         await saveData(accessToken, patchRequestUrl, e.uri, entity, saveDataMap, new TextDecoder(CHARSET).decode(newFileData));
     });
-    return saveDataMap;
 }
 
-function createVirtualFile(portalsFS: PortalsFS, fileName: string, languageCode: string, data: any, portalFileExtension: string, subUri: string) {
+function createVirtualFile(portalsFS: PortalsFS, fileName: string, languageCode: string, data: any, portalFileExtension: string, subUri: string,entityId : string, saveDataAtribute : string ) {
+    const saveEntityDetails = new SaveEntityDetails(entityId, fileName, saveDataAtribute);
```
Comment (ashishchoudhary001) [inline src/web/client/common/remoteFetchProvider.ts:100]: Are we incorrectly saving fileName instead of EntityName in the saveDataMap?

**Review by ashishchoudhary001 — CHANGES_REQUESTED** [review]
saveData implementation is still incorrect. It should pick the entity name from the saveDataMap.

File: src/web/client/common/remoteFetchProvider.ts (hunk 1 above)
Comment (vedg) [reply src/web/client/common/remoteFetchProvider.ts]: Yes fileName and entityname are different - updated it

File: src/web/client/common/remoteFetchProvider.ts [hunk 2]
```diff
@@ -93,13 +94,13 @@ function createContentFiles(result: string, entity: string, queryParamsMap: any,
         vscode.window.showInformationMessage(patchRequestUrl)
         await saveData(accessToken, patchRequestUrl, e.uri, entity, saveDataMap, new TextDecoder(CHARSET).decode(newFileData));
     });
-    return saveDataMap;
 }
 
-function createVirtualFile(portalsFS: PortalsFS, fileName: string, languageCode: string, data: any, portalFileExtension: string, subUri: string) {
+function createVirtualFile(portalsFS: PortalsFS, fileName: string, languageCode: string, data: any, portalFileExtension: string, subUri: string,entityId : string, saveDataAtribute : string ) {
+    const saveEntityDetails = new SaveEntityDetails(entityId, entityName, saveDataAtribute);
```
Comment (ashishchoudhary001) [inline src/web/client/common/remoteFetchProvider.ts:100]: from where are you fetching the entityName? I don't see it being passed in the createVirtualFile function.

File: src/web/client/common/remoteSaveProvider.ts [hunk 3]
```diff
@@ -7,25 +7,18 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import * as vscode from 'vscode';
 import { getHeader } from './authenticationProvider';
-import { columnExtension, FILE_EXTENSION_REGEX } from './constants';
 import { ERRORS, showErrorDialog } from './errorHandler';
+import { SaveEntityDetails } from './portalSchemaInterface';
 
-export async function saveData(accessToken: string, requestUrl: string, fileUri: vscode.Uri, entity: string, saveDataMap: Map<string, string>, value: string) {
+export async function saveData(accessToken: string, requestUrl: string, fileUri: vscode.Uri, entity: string, saveDataMap: Map<string, SaveEntityDetails>, value: string) {
```
Comment (ashishchoudhary001) [inline src/web/client/common/remoteSaveProvider.ts:13]: saveData function should not have entity as a parameter. It should fetch entity from the saveDataMap

**Review by ashishchoudhary001 — APPROVED** [review]

**Pull request merged**